tests/unit/h/schemas/api/group_test.py

```diff
@@ -13,167 +13,168 @@
-
-    def test_it_raises_if_name_too_long(self, schema):
-        with pytest.raises(ValidationError, match="name:.*is too long"):
-            schema.validate({"name": "o" * (GROUP_NAME_MAX_LENGTH + 1)})
-
-    def test_it_raises_if_name_isnt_a_string(self, schema):
-        name = 42
-
-        with pytest.raises(
-            ValidationError, match=f"name: {name} is not of type 'string'"
-        ):
-            schema.validate({"name": name})
 
+class TestGroupAPISchema:
     @pytest.mark.parametrize(
-        "name",
+        "group_authority,data,expected_appstruct",
         [
-            f" {'o' * GROUP_NAME_MIN_LENGTH}",
-            f"{'o' * GROUP_NAME_MIN_LENGTH} ",
-            " " * GROUP_NAME_MIN_LENGTH,
+            (  # An empty dict is valid input data.
+                DEFAULT_AUTHORITY,
+                {},
+                {},
+            ),
+            (  # Valid group name.
+                DEFAULT_AUTHORITY,
+                {"name": "Valid Group Name"},
+                {"name": "Valid Group Name"},
+            ),
+            (  # Valid group description.
+                DEFAULT_AUTHORITY,
+                {"description": "Valid group description."},
+                {"description": "Valid group description."},
+            ),
+            (  # Valid groupid (requires matching third-party authority).
+                "thirdparty.com",
+                {"groupid": "group:1234abcd!~*()@thirdparty.com"},
+                {"groupid": "group:1234abcd!~*()@thirdparty.com"},
+            ),
+            (  # All valid fields at once.
+                "thirdparty.com",
+                {
+                    "name": "Valid Group Name",
+                    "description": "Valid group description.",
+                    "groupid": "group:1234abcd!~*()@thirdparty.com",
+                },
+                {
+                    "name": "Valid Group Name",
+                    "description": "Valid group description.",
+                    "groupid": "group:1234abcd!~*()@thirdparty.com",
+                },
+            ),
+            (  # It ignores non-whitelisted properties.
+                DEFAULT_AUTHORITY,
+                {
+                    "organization": "foobar",
+                    "joinable_by": "whoever",
+                },
+                {},
+            ),
         ],
     )
-    def test_it_raises_if_name_has_leading_or_trailing_whitespace(self, schema, name):
-        with pytest.raises(
-            ValidationError,
-            match="Group names can't have leading or trailing whitespace.",
-        ):
-            schema.validate({"name": name})
-
-    def test_it_validates_with_no_data(self, schema):
-        appstruct = schema.validate({})
-
-        assert appstruct == {}
-
-    def test_it_validates_with_valid_name(self, schema):
-        appstruct = schema.validate({"name": "Perfectly Fine"})
-
-        assert "name" in appstruct
```

Comment:
It's weird that these tests just test that `"name"` is in the returned `appstruct`, but didn't actually test what the value of `"name"` was